Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -1,35 +1,78 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.Collections.Generic;
+
+namespace BootCamp.Chapter
 {
     public class Inventory
     {
-        private Item[] _items;
-        public Item[] GetItems()
-        {
-            return new Item[0];
-        }
+        public List<Item> Items { get; set; }
 
         public Inventory()
         {
-            _items = new Item[0];
+            Items = new List<Item>();
         }
 
-        public Item[] GetItems(string name)
+        /// <summary>
+        /// Returns the item it found based on the name you gave, if no item is found returns null
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public Item GetItem(string name)
         {
-            return new Item[0];
+            NullChecks.StringNullChecks(name);
+
+            foreach (Item item in Items)
+            {
+                if (item?.Name == name)
+                {
+                    return item;
+                }
+            }
+            return null;
         }
+        /// <summary>
+        /// Returns a List of all items that are in the inventory based on the name you gave.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        public List<Item> GetItems(string name)
+        {
+            NullChecks.StringNullChecks(name);
+
+            List<Item> final = new List<Item>();
 
+            for (int i = 0; i < Items.Count; i++)
+            {
+                if (Items[i]?.Name == name)
+                {
+                    final.Add(Items[i]);
+                }
+            }
+            return final;
+        }
+        /// <summary>
+        /// Add an item to the inventory.
+        /// </summary>
+        /// <param name="item"></param>
         public void AddItem(Item item)
         {
-
+            Items.Add(item ?? throw new ArgumentNullException($"{nameof(item)} can't be null."));
         }
 
         /// <summary>
-        /// Removes item matching criteria by item.
-        /// Does nothing if no such item exists
+        /// Removes ONE item matching criteria by item.
+        /// Does nothing if no such item exists.
         /// </summary>
         public void RemoveItem(Item item)
         {
-
+            for (int i = 0; i < Items.Count; i++)
+            {
+                if (Items[i].IsTheSameItem(item ?? throw new ArgumentNullException($"{nameof(item)} can't be null.")))
```

Comment:
i could change this to:
if (Items[i]?.IsTheSameItem(item ?? throw new ArgumentNullException($"{nameof(item)} can't be null.")) ?? false) 
but at the moment items cannot have a null item i think.

Reply:
I think in this case, it's better to be a bit more simple. For homework's sake, keep it, but normally you should "fail early" therefore check the item not being null at the start and especially not at the loop.